bindings/java/src/org/sleuthkit/datamodel/CommunicationsManager.java

```diff
@@ -0,0 +1,337 @@
+		return db.getAccountTypesInUse();
+	}
+
+	/**
+	 * Get all accounts of given type
+	 * 
+	 * @param type account type
+	 * 
+	 * @return List <Account.Type>, list of accounts
+	 *
+	 * @throws TskCoreException exception thrown if a critical error occurs
+	 *                          within TSK core
+	 */
+	public List<Account> getAccounts(Account.Type accountType) throws TskCoreException {
+		
+		List<Account> accounts =  new ArrayList<Account>();
+		
+		List<BlackboardArtifact> artifacts = db.getAccountArtifacts(accountType);
+		for (BlackboardArtifact accountArtifact: artifacts) {
+			accounts.add(new Account(this.db, accountArtifact.getArtifactID() ));
+		}
+	
+		return accounts;
+	}
+
+	
+/**
+ * Reject the given account
+ * 
+ * @param account account to reject
+ * 
+ * @throws TskCoreException exception thrown if a critical error occurs
+ *                          within TSK core
+ */
+public void rejectAccount(Account account) throws TskCoreException {
+	db.setReviewStatus(db.getBlackboardArtifact(account.getArtifactId()), BlackboardArtifact.ReviewStatus.REJECTED);
+}
+
+		
+/**
+ * Get all accounts that have any kind of relationship with the given account
+ * 
+ * @param sender sender account
+ * @paraman recipients list of recipients
+ * @param communicationArtifact communication item
+ * 
+ * @throws TskCoreException exception thrown if a critical error occurs
+ *                          within TSK core
+ */
+	public List<Account> getAccountsWithRelationship(Account account) throws TskCoreException {
+		
+		List<Account> accounts =  new ArrayList<Account>();
+		
+		List<BlackboardArtifact> artifacts = db.getAccountsWithRelationship(account.getArtifactId());
+		for (BlackboardArtifact accountArtifact: artifacts) {
+			accounts.add(new Account(this.db, accountArtifact.getArtifactID() ));
+		}
+	
+		return accounts;
+	}
+
+/**
+ * Add a relationship between the given sender and recipient accounts.
+ * 
+ * @param sender sender account
+ * @paraman recipients list of recipients
+ * @param communicationArtifact communication item
+ * 
+ * @throws TskCoreException exception thrown if a critical error occurs
+ *                          within TSK core
+ */
+public void addRelationships(Account sender, List<Account> recipients, BlackboardArtifact communicationArtifact) {
+	
+	// Currently we do not save the direction of communication
+	List<Long> accountIDs = new ArrayList<Long>();
+	if (null != sender) {
+		accountIDs.add(sender.getArtifactId());
+	}
+	
+	for (Account recipient: recipients) {
+		accountIDs.add(recipient.getArtifactId());
+	}
+	
+	Set<UnorderedPair<Long>> relationships = listToUnorderedPairs(accountIDs);
+	
+	Iterator<UnorderedPair<Long>> iter = relationships.iterator();
+	
+	while (iter.hasNext()) {
+		
+		try {
+			UnorderedPair<Long> accountPair = iter.next();
+			if (accountPair.first.longValue() != accountPair.second.longValue()) {// dont add a relationship with self
```

Comment:
What if a relationship with self is what is intended such as an email to oneself?  Gmail jsut let me send an email to myself.